lib/vector-buffers/src/variants/disk_v2/common.rs

```diff
@@ -39,10 +44,57 @@ pub const MAX_FILE_ID: u16 = u16::MAX;
 #[cfg(test)]
 pub const MAX_FILE_ID: u16 = 6;
 
+// The alignment used by the record serializer.
+const SERIALIZER_ALIGNMENT: usize = 16;
+const MAX_ALIGNABLE_AMOUNT: usize = usize::MAX - SERIALIZER_ALIGNMENT;
+
 pub(crate) fn create_crc32c_hasher() -> Hasher {
     crc32fast::Hasher::new()
 }
 
+/// Aligns the given amount to 16.
+///
+/// This is required due to the overalignment used in record serialization, such that we can correctly determine minimum
+/// on-disk sizes for various elements, and account for those in size limits, etc.
+pub(crate) const fn align16(amount: usize) -> usize {
+    // The amount must be less than `MAX_ALIGNABLE_AMOUNT` otherwise we'll overflow trying to align it, ending up with a
+    // nonsensicial value.
+    assert!(
+        amount <= MAX_ALIGNABLE_AMOUNT,
+        "`amount` must be less than `MAX_ALIGNABLE_AMOUNT`"
+    );
+
+    ((amount + SERIALIZER_ALIGNMENT - 1) / SERIALIZER_ALIGNMENT) * SERIALIZER_ALIGNMENT
```

Comment:
I just checked the assembly output for this, and LLVM translates this internally to `add 15` then `and ~15`. The other variant I suggested translates to the same. Smart.

Reply:
Yeah, I figured that the comments in the unit/property test -- avoiding depending on round to zero behavior, etc -- were going to be more of a very niche cover-our-ass situation, and that LLVM would do something fairly smart/well-defined given the power-of-two nature of the alignment amount. Good to know it's _definitely_ doing that at the moment.